dark/filter.py

```diff
@@ -116,6 +116,93 @@ def accept(self, title):
         return self.DEFAULT_ACCEPT
 
 
+class SequenceFilter:
+    """
+    Provide an acceptance test for sequences.
+
+    @param whitelist: If not C{None}, a C{set} of exact sequences that are always
+        acceptable.
+    @param blacklist: If not C{None}, a C{set} of exact sequences that are never
+        acceptable.
+    @param whitelistFile: If not C{None}, a C{str} filename containing lines
+        that give exact sequences that are always acceptable.
+    @param blacklistFile: If not C{None}, a C{str} filename containing lines
+        that give exact sequences that are never acceptable.
+    @param positiveRegex: If not C{None}, a C{str} regex that sequences
+        must match (case is ignored).
+    @param negativeRegex: If not C{None}, a C{str} regex that sequences
+        must not match (case is ignored).
+    """
+
+    REJECT = 0
+    WHITELIST_ACCEPT = 1
+    DEFAULT_ACCEPT = 2
+
+    def __init__(
+        self,
+        whitelist=None,
+        blacklist=None,
+        whitelistFile=None,
+        blacklistFile=None,
+        positiveRegex=None,
+        negativeRegex=None,
+    ):
+        whitelist = whitelist or set()
```

Comment:
Make sure `whitelist` is a set, either by type hint or casting.

Reply:
Done, and I added type hints to several other functions.